### PR TITLE
[#5370] feat(iceberg):  make IcebergTableOperationDispatcher interface more extendable

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?

add `IcebergRequestContext` in `IcebergTableOperationDispatcher`

### Why are the changes needed?

Fix: #5370 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by puchengy** [comment]
Do we plan to make similar changes to `IcebergNamespaceOperations` and `IcebergViewOperations` once those classes are switched to use `getCatalogWrapper` method?

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java [hunk 1]
```diff
@@ -74,17 +75,18 @@ public IcebergTableEventDispatcher(
...
-      String catalogName, Namespace namespace, CreateTableRequest createTableRequest) {
+      IcebergRequestContext context, Namespace namespace, CreateTableRequest createTableRequest) {
     TableIdentifier tableIdentifier = TableIdentifier.of(namespace, createTableRequest.name());
     NameIdentifier nameIdentifier =
-        IcebergRestUtils.getGravitinoNameIdentifier(metalakeName, catalogName, tableIdentifier);
+        IcebergRestUtils.getGravitinoNameIdentifier(
+            metalakeName, context.getCatalogName(), tableIdentifier);
     eventBus.dispatchEvent(
         new IcebergCreateTablePreEvent(
             PrincipalUtils.getCurrentUserName(), nameIdentifier, createTableRequest));
```
Comment (puchengy) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java:85]: Would it be possible to pass the whole context object inside pre-listener and post listener?

**Comment by FANNG1** [comment]
> Do we plan to make similar changes to `IcebergNamespaceOperations` and `IcebergViewOperations` once those classes are switched to use `getCatalogWrapper` method?

yes, will do

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java (hunk 1 above)
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java:85]: Could you describe the scenarios?

Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java:85]: Sounds reasonable, @jerryshao WDYT?

Comment (puchengy) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java:85]: @jerryshao Hi Jerry, I think your idea works for filter checking. What about the use case of headers logging along with table events? How would that look like? Thanks.

Comment (jerryshao) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java:85]: Hi @puchengy how about we merge this PR first, and discuss about your case in a separate issue?

**Review by jerryshao — APPROVED** [review]

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java (hunk 1 above)
Comment (puchengy) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java:85]: @jerryshao SG, thank you.

**Pull request merged**